api/compute/v1alpha1/machinereservation_types.go

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	corev1alpha1 "github.com/ironcore-dev/ironcore/api/core/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ReservationSpec defines the desired state of Reservation
+type ReservationSpec struct {
+	Pools     []corev1.LocalObjectReference `json:"pools"`
+	Resources corev1alpha1.ResourceList     `json:"capabilities,omitempty"`
+}
+
+// ReservationStatus defines the observed state of Reservation
+type ReservationStatus struct {
+	Pools []ReservationPoolStatus `json:"pools,omitempty"`
```

Comment:
the `pools` field should be a `map[string]ReservationPoolStatus` where the key value is the machinepool name. This enables a much easier approach to avoid duplication in the tracked items.